### ISSUE TITLE
unit-case

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1025]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

New rule `unit-quantities-case`. Check case of units. Options: "lower"|"upper"

Example:

```css
a {
  margin: 10PX;
}
```

```css
a {
  margin: 10px;
}
```

**Comment by davidtheclark** [comment]
Can you confirm: are both of those *valid*?

**Comment by evilebottnawi** [comment]
@davidtheclark You can write in upper and lower case. Both valid.

**Comment by jeddy3** [comment]
Oh wow, who knew? ["Like keywords, unit identifiers are ASCII case-insensitive."](https://www.w3.org/TR/css3-values/#dimensions). I kind of want to stick caps-lock on and shout my units from this point forward :)

As such, this seems like a legitimate rule for core to me.

How does the following sound?:

- Name: `unit-case`
- Primary options: `"upper"|"lower"`
- Secondary options: none
- Message: `Expected "${actual}" to be "${expected}" (unit-case)`.

**Comment by evilebottnawi** [comment]
@jeddy3 sound good

**Comment by jeddy3** [comment]
> because i created 2 issue.

Sgtm. We've always tried to keep the rules lazer-focused.

**Comment by evilebottnawi** [comment]
@jeddy3 if the rule is adopted, I do PR tomorrow (we have many `px` and `PX` in our styles in project and I would like to begin to control it :smile: )

**Comment by jeddy3** [comment]
@davidtheclark Are you happy with the inclusion of these `*-case` rules into core? If so, I'll label them up accordingly so @evilebottnawi can get stuck in.

**Comment by davidtheclark** [comment]
Yeah sure!

**Comment by evilebottnawi** [comment]
Merged. Close.
